lib/internal/http2/core.js

```diff
@@ -1675,13 +1657,10 @@ class Http2Stream extends Duplex {
     const handle = this[kHandle];
     const req = new WriteWrap();
     req.stream = this[kID];
-    req.handle = handle;
     req.callback = cb;
-    req.oncomplete = afterDoStreamWrite;
-    req.async = false;
-    const err = createWriteReq(req, handle, data, encoding);
-    if (err)
-      throw errnoException(err, 'write', req.error);
```

Comment:
The semantics here are now different. We're calling destroy with an error instead of just throwing. IMO we should just restore the `if (err)` blocks and not have that code inside the helpers.

Reply:
Actually, I see now that @addaleax requested this. Since this would only destroy the stream, I suppose that should be fine.

Reply:
@apapirovski Fwiw, this is taken from https://github.com/nodejs/node/pull/19389, so that one should land first